packages/api/lambdas/internal-reconciliation-report.js

```diff
@@ -0,0 +1,278 @@
+  let [nextEsItem, nextDbItem] = await Promise.all([esGranulesIterator.peek(), dbGranulesIterator.peek()]); // eslint-disable-line max-len
+
+  /* eslint-disable no-await-in-loop */
+  while (nextEsItem && nextDbItem) {
+    if (nextEsItem.granuleId < nextDbItem.granuleId) {
+      // Found an item that is only in ES and not in DB
+      onlyInEs.push(pick(nextEsItem, granuleFields));
+      await esGranulesIterator.shift();
+    } else if (nextEsItem.granuleId > nextDbItem.granuleId) {
+      // Found an item that is only in DB and not in ES
+      onlyInDb.push(pick(nextDbItem, granuleFields));
+      await dbGranulesIterator.shift();
+    } else {
+      // Found an item that is in both ES and DB
+      if (isEqual(omit(nextEsItem, fieldsIgnored), omit(nextDbItem, fieldsIgnored))) {
+        okCount += 1;
+      } else {
+        withConflicts.push({ es: nextEsItem, db: nextDbItem });
+      }
+      await Promise.all([esGranulesIterator.shift(), dbGranulesIterator.shift()]);
+    }
+
+    [nextEsItem, nextDbItem] = await Promise.all([esGranulesIterator.peek(), dbGranulesIterator.peek()]); // eslint-disable-line max-len
+  }
+
+  // Add any remaining ES items to the report
+  while (await esGranulesIterator.peek()) {
+    const item = await esGranulesIterator.shift();
+    onlyInEs.push(pick(item, granuleFields));
+  }
+
+  // Add any remaining DB items to the report
+  while (await dbGranulesIterator.peek()) {
+    const item = await dbGranulesIterator.shift();
+    onlyInDb.push(pick(item, granuleFields));
+  }
+  /* eslint-enable no-await-in-loop */
+
+  return { okCount, withConflicts, onlyInEs, onlyInDb };
+}
+
+/**
+ * Compare the granule holdings in Elasticsearch with Database
+ *
+ * @param {Object} recReportParams - lambda's input filtering parameters to
+ *                                   narrow limit of report.
+ * @returns {Promise<Object>} an object with the okCount, onlyInEs, onlyInDb
+ * and withConfilcts
+ */
+async function reconciliationReportForGranules(recReportParams) {
+  log.debug('internal-reconciliation-report reconciliationReportForGranules');
+  // To avoid 'scan' granules table, we query GSI in granules table with collectionId.
+  // compare granule holdings:
+  //   Get all collections list from db and es or use the collectionId from the request
+  //   For each collection,
+  //     compare granule holdings and get report
+  //   Report granules only in ES
+  //   Report granules only in DynamoDB
+  //   Report granules with different contents
+
+  const { collectionId, ...searchParams } = recReportParams;
+  // get collections list in ES and dynamoDB combined
+  const collections = collectionId ? [collectionId] : await getAllCollections();
+
+  const concurrencyLimit = process.env.CONCURRENCY || 3;
+  const limit = pLimit(concurrencyLimit);
+
+  const reports = await Promise.all(collections.map((collection) => limit(() =>
+    reportForGranulesByCollectionId(collection, searchParams))));
```

Comment:
```suggestion
  const reports = await Promise.all(collectionIds.map((collId) => limit(() =>
    reportForGranulesByCollectionId(collId, searchParams))));
```

Reply:
I'd make it collectionId, but that results in a lint error for upper scope variable.